test/system/120-load.bats

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bats   -*- bats -*-
+#
+# tests for podman load
+#
+
+load helpers
+
+# Custom helpers for this test only. These just save us having to duplicate
+# the same thing four times (two tests, each with -i and stdin).
+#
+# initialize, read image ID and name
+get_iid_and_name() {
+    run_podman images --format '{{.ID}} {{.Repository}}:{{.Tag}}'
+    read iid img_name < <(echo "$output")
+
+    archive=$PODMAN_TMPDIR/myimage-$(random_string 8).tar
+}
+
+# Simple verification of image ID and name
+verify_iid_and_name() {
+    run_podman images --format '{{.ID}} {{.Repository}}:{{.Tag}}'
+    read new_iid new_img_name < <(echo "$output")
+
+    # Verify
+    is "$new_iid"      "$iid" "Image ID of loaded image == original"
+    is "$new_img_name" "$1"   "Name & tag of restored image"
+}
+
+
+@test "podman load - by image ID" {
+    # FIXME: how to build a simple archive instead?
```

Comment:
I meant, including a pre-created simple image as part of the tests. I'm leaning toward not-worth-it but not convinced yet.

Reply:
Ahh, hmmm.  Ya, I'm on the fence on that as well.  The e2e tests do it out of necessity, Docker-Autotest did it, but that's a confused system and integration suite.

It would be good, from the perspective of avoiding "testing" functioning of external repositories, we can rely on others do do that.  It's bad from the perspective of having additional stuff (here) to maintain.

One idea could be...re-use some/more of the buildah BATS tests as a pre-requesite to ``001-basic.bats``?  As in, download one of them and execute it.  Sure it's a bit of reliance on a side-effect (producing the test image(s) required).  However, I think this might be safe, since if buildah breaks, everything else down the line is guaranteed to break :grinning:

Reply:
I've just reviewed the buildah tests and am not sure I understand your suggestion. But in any case, much as I'd like to eliminate the dependency on the network and remote registries... I don't think it'll be practical.

Reply:
> much as I'd like to eliminate the dependency on the network and remote registries

In general it's good to avoid it as a dependency, but in practice impossible to eliminate.  I was thinking more in terms eating that cost once for multiple tests instead of possibly having lots of tests constantly pulling down the same image over and over (and be more likely to break).  Since buildah tests images, I thought perhaps it might already have something we could borrow.  If not, no worries, this isn't something that needs solving immediately.  Thanks for checking :smile: